### PR TITLE
Python3: fix remaining syntax errors found by py2exe

### PR DESCRIPTION
**Pull request by michaelDCurran**
### Link to issue number:
None.

### Summary of the issue:
When running py2exe over the codebase, it found some syntax errors and one missing module.
 
### Description of how this pull request fixes the issue:
Fixes all remaining syntax errors:
* tuples in a for loop need to be in brackets.
* One file had inconsistant whitepsace where there was a space at the beginning of a line before tabbed indentation.
* numbers no longer need or can have an 'L' suffix at the end.
* the 'ur' string prefix is no longer supported. Use 'r' instead. Note this was taken from draft pr #9636 plus further strings missed by that pr.
 * Several strings containing backslashes did not have an 'r' prefix when they should have.
* collections.Sequence is now collections.abc.Sequence.
* raise excType,value,traceback is no longer valid syntax. Changed the one place we did this to just hold and raise the original exception object, as Python3 now includes the traceback on the exception object itself.

### Testing performed:
Py2exe successfully parsed all modules in the codebase.

### Known issues with pull request:
Py2exe still fails later on, unrelated to these syntax errors. See my next comment.

### Change log entry:
None.

### CODE REVIEW

**Comment by michaelDCurran** [comment]
Py2exe still fails later on when copying py2exe/pywintypes32.dll. 
It throws an assertion error because although the module has a ```__file__```, attribute, its file extension is not one of the accepted python extensions. Rather it is .dll.
I'm not sure if this is a bug in py2exe... @leonardder or @albertosottile ?   
I saw that it was pywintypes37.dll by adding
```
print(mod.__file__)
```
directly above the assertion. 
Full traceback follows:
```
Traceback (most recent call last):
  File "setup.py", line 231, in <module>
    + getRecursiveDataFiles('documentation', '../user_docs', excludes=('*.t2t', '*.t2tconf', '*/developerGuide.*'))
  File "C:\Program Files (x86)\Python37-32\lib\site-packages\setuptools\__init__.py", line 145, in setup
    return distutils.core.setup(**attrs)
  File "C:\Program Files (x86)\Python37-32\lib\distutils\core.py", line 148, in setup
    dist.run_commands()
  File "C:\Program Files (x86)\Python37-32\lib\distutils\dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "C:\Program Files (x86)\Python37-32\lib\distutils\dist.py", line 985, in run_command
    cmd_obj.run()
  File "setup.py", line 104, in run
    super(py2exe, self).run()
  File "C:\Users\mick\programming\git\nvda\include\py2exe\py2exe\distutils_buildexe.py", line 192, in run
    self._run()
  File "C:\Users\mick\programming\git\nvda\include\py2exe\py2exe\distutils_buildexe.py", line 273, in _run
    builder.build()
  File "C:\Users\mick\programming\git\nvda\include\py2exe\py2exe\runtime.py", line 264, in build
    self.copy_files(destdir)
  File "C:\Users\mick\programming\git\nvda\include\py2exe\py2exe\runtime.py", line 508, in copy_files
    assert mod.__file__.endswith(EXTENSION_SUFFIXES[0])
AssertionError
```

**Comment by leonardder** [comment]
In my opinion, the actual bug here is that py2exe tries to copy stuff for pywin32, but we don't use pywin32 anymore within our code base. So rather than coming up with a fix or workaround for the assertion, we should try to stop copying parts of pywin32 altogether.

**Comment by leonardder** [comment]
See #9711 and #9701

File: source/watchdog.py [hunk 1]
```diff
@@ -291,7 +291,11 @@ def execute(self, func, args, kwargs, pumpMessages=True):
 
 		exc = self._exc_info
 		if exc:
-			raise exc[0], exc[1], exc[2]
+			# The execution of the function in the other thread cuased an exception.
+			# Re-raise it here.
+			# Note that in Python3, the traceback (stack) is now part of the exception,
+			# So the logged traceback will correctly show the stack for both this thread and the other thread. 
+			raise e
```
Comment (leonardder) [inline source/watchdog.py]: Where is this e  coming from? Shouldn't this be exc?

**Review by LeonarddeR — CHANGES_REQUESTED** [review]

**Comment by michaelDCurran** [comment]
@leonardder  I believe I have addressed your review comments. Changes of basestring etc are coming in a separate pr very soon.
I have removed the accidentally committed test_traceback.py ;) but at least you got to see how I confirmed how Python3 exceptions work.

**Comment by michaelDCurran** [comment]
setup.py now excludes win32api (which we don't need), so it gets around the possible bug when trying to include pywintypes37.dll.

File: source/setup.py [hunk 2]
```diff
@@ -190,7 +190,15 @@ def getRecursiveDataFiles(dest,source,excludes=()):
...
 		"excludes": ["tkinter",
-			"serial.loopback_connection", "serial.rfc2217", "serial.serialcli", "serial.serialjava", "serial.serialposix", "serial.socket_connection"],
+			"serial.loopback_connection", 
+			"serial.rfc2217", 
+			"serial.serialcli", 
+			"serial.serialjava", 
+			"serial.serialposix", 
+			"serial.socket_connection",
+			# Py2exe seems to implicitly include this, but we don't need it. 
+			"win32api",
```
Comment (leonardder) [inline source/setup.py]: I'd really like to know why this was included.

Also note the following lines from the output:

```
Copy d:\Development\NVDA\source repo\include\py2exe\win32wnet.pyd to D:\Development\NVDA\source repo\dist\win32wnet.pyd
Copy d:\Development\NVDA\source repo\include\py2exe\_winxptheme.pyd to D:\Development\NVDA\source repo\dist\_winxptheme.pyd
Copy ExtensionDLL d:\Development\NVDA\source repo\include\py2exe\pywintypes37.dll to D:\Development\NVDA\source repo\dist\
```

These 3 files are all pywin32. We also have the following files in library.zip:

* win32wnet.pyc
* winxptheme.pyc
* _winxptheme.pyc

**Review by LeonarddeR — APPROVED** [review]

**Comment by leonardder** [comment]
Small additional detail. A non public application I wrote and converted to py2exe lately does not contain all these modules. So I really have the feeling that there is a reference in our code we aren't currently aware of.

**Comment by michaelDCurran** [comment]
If I manually delete netbios.py, win32api.pyd, win32wnet.pyd and 
winxptheme.py from our py2exe git submodule,

Py2exe then tells me the following modules are missing and why:

? netbios                             imported from uuid

? winxptheme                          imported from 
wx.lib.agw.aui.dockart, wx.lib.agw.aui.framemanager

? winxptheme                          imported from 
wx.lib.agw.aui.dockart, wx.lib.agw.aui.framemanager


I have confirmed online that winxptheme (from pywin32 is used in 
wxPython if available.


I can also see from Python3's own uuid module source code, that to 
generate a uuid, it  tries various different ways of generating a 
hardware node value, one of those ways is to use netbios / win32wnet if 
available.


In short, we should be safe to exclude winxptheme, netbios and win32wnet.


However, @leonardder, what was the reasoning behind why you thought that 
py2exe depended upon pywin32? I assume it was something more significant 
than missing modules at this stage in the build?

I.e. knowing this now, should we just put these in excludes, or can we 
actually remove these modules from the git submodule?
